components/brave_rewards/browser/diagnostic_log.cc

```diff
@@ -0,0 +1,365 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/browser/diagnostic_log.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/files/file_util.h"
+#include "base/i18n/time_formatting.h"
+#include "base/strings/stringprintf.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/task/post_task.h"
+#include "base/task/task_traits.h"
+
+namespace {
+
+const int64_t kChunkSize = 1024;
+const size_t kDividerLength = 80;
+
+std::string FormatTime(const base::Time& time) {
+  return base::UTF16ToUTF8(
+      base::TimeFormatWithPattern(time, "MMM dd, YYYY h::mm::ss.S a"));
+}
+
+std::string GetLogVerboseLevelName(const int verbose_level) {
+  std::string verbose_level_name;
+
+  switch (verbose_level) {
+    case 0: {
+      verbose_level_name = "ERROR";
+      break;
+    }
+
+    case 1: {
+      verbose_level_name = "INFO";
+      break;
+    }
+
+    default: {
+      verbose_level_name = "VERBOSE";
+      verbose_level_name += std::to_string(verbose_level);
+      break;
+    }
+  }
+
+  return verbose_level_name;
+}
+
+bool Create(const base::FilePath& file_path, base::File* file) {
+  DCHECK(file);
+
+  file->Initialize(file_path, base::File::FLAG_CREATE_ALWAYS |
+                                  base::File::FLAG_READ |
+                                  base::File::FLAG_WRITE);
+
+  return file->IsValid();
+}
+
+bool Open(const base::FilePath& file_path, base::File* file) {
+  DCHECK(file);
+
+  if (file->IsValid()) {
```

Comment:
It seems that for all callers, `IsValid` should be false, so it may be more appropriate to remove this branch and just let `file->Initialize` DCHECK if it's valid. Also OK to leave it alone, though!

Reply:
Good idea, will fix.